src/gui/ogr/qgsnewogrconnection.cpp

```diff
@@ -54,11 +54,18 @@ QgsNewOgrConnection::QgsNewOgrConnection( QWidget *parent, const QString &connTy
   const QStringList dbDrivers = QgsProviderRegistry::instance()->databaseDrivers().split( ';' );
   for ( int i = 0; i < dbDrivers.count(); i++ )
   {
-    const QString dbDrive = dbDrivers.at( i );
-    cmbDatabaseTypes->addItem( dbDrive.split( ',' ).at( 0 ) );
+    const QString dbDriver = dbDrivers.at( i );
+    if ( ( !dbDriver.isEmpty() ) && ( !dbDriver.isNull() ) )
```

Comment:
```suggestion
    if ( !dbDriver.isEmpty() )
```

It's not possible for this string to be null

Reply:
https://github.com/qgis/QGIS/blob/8491e83166ea173fe74ef046dfed0f0f7dbc2e0c/src/gui/providers/ogr/qgsogrsourceselect.cpp#L81-L82
https://github.com/qgis/QGIS/blob/8491e83166ea173fe74ef046dfed0f0f7dbc2e0c/src/gui/providers/ogr/qgsogrsourceselect.cpp#L90-L91
https://github.com/qgis/QGIS/blob/8491e83166ea173fe74ef046dfed0f0f7dbc2e0c/src/gui/providers/ogr/qgsogrsourceselect.cpp#L101-L102
should also be fixed?

Reply:
Yes, there's no way QString::split will return a null string. It'll always be empty.

Reply:
Done.